storage/google/cloud/storage/blob.py

```diff
@@ -688,6 +691,9 @@ def _do_multipart_upload(self, client, stream, content_type,
 
         if self.user_project is not None:
             name_value_pairs.append(('userProject', self.user_project))
+        if predefined_acl is not None:
+            self._acl.save_predefined(predefined_acl)
+            name_value_pairs.append(('predefinedAcl', predefined_acl))
```

Comment:
I don't think we can use `save_predefined` here -- it forces an API request, which will fail for not-yet-existing blobs.  We likely need to factor out the "name conversion" logic from that method for reuse here.

A system test which exercises the feature would have made this clear.

Reply:
Also, this argument shouldn't need to be passed down from `upload_*` to the helper methods:  instead, the top-level methods should do the work of setting the ACL:  the lower-level methods will generate the right payload from that.